frontend/rust-lib/flowy-database2/src/services/database_view/view_editor.rs

```diff
@@ -508,6 +530,78 @@ impl DatabaseViewEditor {
     Ok(())
   }
 
+  pub async fn v_get_all_calculations(&self) -> Vec<Arc<Calculation>> {
+    self.delegate.get_all_calculations(&self.view_id)
+  }
+
+  pub async fn v_update_calculations(
+    &self,
+    params: UpdateCalculationChangesetPB,
+  ) -> FlowyResult<()> {
+    let calculation_id = match params.calculation_id {
+      None => gen_database_calculation_id(),
+      Some(calculation_id) => calculation_id,
+    };
+
+    let calculation = Calculation {
```

Comment:
How about create a method in the Calculation. Like `Calculation::Non(id, field_id, calculation_type)`